### PR TITLE
files_trashbin: Add OpenAPI spec

### PR DESCRIPTION
**Pull request by provokateurin**
## Summary

From https://github.com/nextcloud/server/pull/36666. Adds the necessary annotations and descriptions.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/files_trashbin/openapi.json [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                ],
+                "parameters": [
+                    {
+                        "name": "fileId",
+                        "in": "query",
+                        "description": "ID of the file",
+                        "schema": {
+                            "type": "integer",
+                            "format": "int64",
+                            "default": 1
```
Comment (bigcat88) [inline apps/files_trashbin/openapi.json:68]: maybe `-1`?

**Review by bigcat88 — CHANGES_REQUESTED** [review]

File: apps/files_trashbin/openapi.json (hunk 1 above)
Comment (provokateurin) [reply apps/files_trashbin/openapi.json]: Fixed it, there was a missing minus when parsing the default value.

**Review by bigcat88 — APPROVED** [review]

File: apps/files_trashbin/openapi.json [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+                            "default": 32
+                        }
+                    },
+                    {
+                        "name": "a",
+                        "in": "query",
+                        "description": "Whether to crop the preview",
+                        "schema": {
+                            "type": "integer",
+                            "default": 0
```
Comment (Pytal) [inline apps/files_trashbin/openapi.json:97]:
```suggestion
                            "type": "boolean",
                            "default": false
```

File: apps/files_trashbin/openapi.json [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                        "schema": {
+                            "type": "integer",
+                            "format": "int64",
+                            "default": 32
+                        }
+                    },
+                    {
+                        "name": "a",
+                        "in": "query",
+                        "description": "Whether to crop the preview",
```
Comment (Pytal) [inline apps/files_trashbin/openapi.json:94]:
```suggestion
                        "description": "Whether to not crop the preview",
```

File: apps/files_trashbin/openapi.json (hunk 2 above)
Comment (provokateurin) [reply apps/files_trashbin/openapi.json:97]: This is due to a quirk how booleans are handled by the server. Any value will result in `true`, except when sending a `0` which will turn into `false`. In theory [`allowEmptyValue`](https://swagger.io/docs/specification/describing-parameters/#nullable) exists and could be used to describe this case with the value being a boolean, but none of the generators that are widely used implement it (https://github.com/search?q=repo%3AOpenAPITools%2Fopenapi-generator%20allowEmptyValue&type=code). For the clients that are generated with them all boolean parameters would be broken, that's why the integer variant is used here.

**Review by skjnldsv — APPROVED** [review]

**Review by Pytal — APPROVED** [review]

**Pull request merged**